src/Client.ts

```diff
@@ -93,7 +93,7 @@ export class Client {
 		} else if (clientOptions.authProvider !== undefined) {
 			httpClient = HTTPClientFactory.createWithAuthenticationProvider(clientOptions.authProvider);
 		} else if (clientOptions.middleware !== undefined) {
-			httpClient = new HTTPClient(clientOptions.middleware);
+			httpClient = new HTTPClient(...[].concat(clientOptions.middleware));
```

Comment:
@ddyett  Thank you for your help with the code for integrating the spread operator!

Reply:
@nikithauc @ddyett great workaround! Thanks for taking the time to integrate the suggestions in the code base!